src/server/tiered_storage.cc

```diff
@@ -355,6 +356,68 @@ void TieredStorage::Free(PrimeIterator it, DbTableStats* stats) {
   stats->tiered_size -= len;
 }
 
+size_t TieredStorage::GetBinSize(size_t blob_len) {
+  CHECK(blob_len <= kMaxSmallBin);
+  unsigned bin_index = SmallToBin(blob_len);
+  return kSmallBins[bin_index];
+}
+
+void TieredStorage::Defrag(DbIndex db_index) {
+  // start scanning from a random position in the ref_cnt_ table.
```

Comment:
why do we want to start in a random position?

Reply:
since if the first few entries never change and they don't have fragmentation, then we will never be able to collect others.